flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskAsyncCallTest.java

```diff
@@ -289,7 +290,7 @@ public void invoke() throws Exception {
 
 			// wait forever (until canceled)
 			synchronized (this) {
-				while (error == null && lastCheckpointId < numCalls) {
```

Comment:
The only locations where a notify was called is when the error is set to `!= null`. Furthermore I think, the testing task is not intended to complete in case that `lastCheckpointId >= numCalls`. Alternatively, one could fix the problem by executing the `lastCheckpointId` under the lock as well as the trigger of `awaitLatch`.